### PR TITLE
Yts-2 Adding validations to models and seed data

### PR DESCRIPTION
**Pull request by Tayyab-athar-1998**

### CODE REVIEW

File: app/models/movie.rb [hunk 1]
```diff
@@ -14,4 +14,19 @@ class Movie < ApplicationRecord
...
+  validate :release_date_is_date
+  validates :synopsis, presence: true
+  validates :languages, presence: true, inclusion: {
+    in: %w[english English Hindi hindi],
+    message: '%<value> is not a allowed language'
+  }
+
+  private
+
+  def release_date_is_date
```
Comment (IDaniyal) [inline app/models/movie.rb]:
```suggestion
  def valid_release_date?
```

**Review by IDaniyal — APPROVED** [review]

File: app/models/movie.rb [hunk 2]
```diff
@@ -9,6 +9,21 @@ class Movie < ApplicationRecord
 
   belongs_to :created_by, class_name: :User, inverse_of: 'movies_created', foreign_key: 'user_id'
 
+  validates :name, presence: true
+  validates :release_date, presence: true
+  validate :release_date_is_date
+  validates :synopsis, presence: true
+  validates :languages, presence: true, inclusion: {
```
Comment (usmansaleem001) [inline app/models/movie.rb:20]: might need to revisit this.

File: app/models/movie_role.rb [hunk 3]
```diff
@@ -1,4 +1,12 @@
 class MovieRole < ApplicationRecord
+  enum allowed_roles: %w[actor director]
```
Comment (usmansaleem001) [inline app/models/movie_role.rb]:
```suggestion
  enum roles: %w[actor director]
```

File: app/models/user.rb [hunk 4]
```diff
@@ -12,4 +12,7 @@ class User < ApplicationRecord
   
   has_many :movies, through: :likes, source: :likeable, source_type: 'Movie'
   has_many :comments, through: :likes, source: :likeable, source_type: 'Comment'
```
Comment (usmansaleem001) [inline app/models/user.rb:14]: read about counter cache and implement here.

**Review by usmansaleem001 — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 4 above)
Comment (Tayyab-athar-1998) [reply app/models/user.rb:14]: implemented it for normal associations looking into how to implement it for polymorphic associations

**Comment by Tayyab-athar-1998** [comment]
TODO:
1- Style admin panel views and move them from erb to haml.
2- Look into counter caching for polymorphic association and implement it in.
3- Remove unused actions from movie resource after it movie page implementation is complete.

**Pull request merged**